app/controlplane/internal/service/auth.go

```diff
@@ -258,33 +274,34 @@ func (c *upstreamOIDCclaims) preferredEmail() string {
 
 func callbackHandler(svc *AuthService, w http.ResponseWriter, r *http.Request) *oauthResp {
 	ctx := context.Background()
-	// if OIDC provider returns an error, return it to and show it to the user
-	if desc := r.URL.Query().Get("error_description"); desc != "" {
-		return &oauthResp{http.StatusUnauthorized, errors.New(desc), true}
+	// if OIDC provider returns an error, redirect to the login page to and show it to the user
+	if desc := r.URL.Query().Get(oidcErrorParam); desc != "" {
+		redirectURL := fmt.Sprintf("%s?%s=%s", svc.AuthURLs.Login, oidcErrorParam, desc)
```

Comment:
I'm not sure if we can use here `url.Parse` and build the final URL. I'm not sure if it needs to be encoded.

See this example: https://go.dev/play/p/YCTvdluws-r

Reply:
Yes, makes sense. I'll encode the parameter value.